test/e2e/app-dir/use-cache/app/(dynamic)/hoc/[slug]/with-slug.tsx

```diff
@@ -1,7 +1,10 @@
-export function withSlug<P extends { params: Promise<{ slug: string }> }>(
+export function withSlug(
   Component: React.ComponentType<{ slug: string }>
-): React.ComponentType<{ params: Promise<{ slug: string }> }> {
-  return async function ComponentWithSlug(props: P) {
```

Comment:
This only worked because `strict: false`. The claim was to return something more narrower than `P`. The implementation doesn't even consider the generic type. It only accepts a single prop